spec/filters/geoip_ecs_spec.rb

```diff
@@ -0,0 +1,203 @@
+            expect( event.get ecs_select[disabled: "[#{target}][as_org]", v1: "[#{target}][as][organization][name]"] ).to eq "Google LLC"
+          end
+        end
+
+        context "with customize fields" do
+          let(:fields) { ["continent_name", "timezone"] }
+          let(:options) { common_options.merge({"fields" => fields}) }
+
+          it "should return fields" do
+            plugin.filter(event)
+
+            expect( event.get ecs_select[disabled: "[#{target}][ip]", v1: "[#{target}][ip]"] ).to be_nil
+            expect( event.get ecs_select[disabled: "[#{target}][country_code2]", v1: "[#{target}][geo][country_iso_code]"] ).to be_nil
+            expect( event.get ecs_select[disabled: "[#{target}][country_name]", v1: "[#{target}][geo][country_name]"] ).to be_nil
+            expect( event.get ecs_select[disabled: "[#{target}][continent_code]", v1: "[#{target}][geo][continent_code]"] ).to be_nil
+            expect( event.get ecs_select[disabled: "[#{target}][location][lat]", v1: "[#{target}][geo][location][lat]"] ).to be_nil
+            expect( event.get ecs_select[disabled: "[#{target}][location][lon]", v1: "[#{target}][geo][location][lon]"] ).to be_nil
+
+            expect( event.get ecs_select[disabled: "[#{target}][continent_name]", v1: "[#{target}][geo][continent_name]"] ).to eq "North America"
+            expect( event.get ecs_select[disabled: "[#{target}][timezone]", v1: "[#{target}][geo][timezone]"] ).to eq "America/Chicago"
+          end
+        end
+
+      end
+    end
+
+    context "setup target field" do
+      let(:ip) { "8.8.8.8" }
+      let(:event) { LogStash::Event.new("message" => ip) }
+      let(:common_options) { {"source" => "message", "database" => CITYDB} }
+
+      context "ECS disabled" do
+        before do
+          allow_any_instance_of(described_class).to receive(:ecs_compatibility).and_return(:disabled)
+          plugin.register
+          plugin.filter(event)
+        end
+
+        context "`target` is unset" do
+          let(:options) { common_options }
+          it "should use 'geoip'" do
+            expect( event.get "[geoip][ip]" ).to eq ip
+          end
+        end
+
+        context "`target` is set" do
+          let(:target) { 'host' }
+          let(:options) { common_options.merge({"target" => target}) }
+          it "should use `target`" do
+            expect( event.get "[#{target}][ip]" ).to eq ip
+          end
+        end
+      end
+
+      context "ECS mode" do
+        before do
+          allow_any_instance_of(described_class).to receive(:ecs_compatibility).and_return(:v1)
+        end
+
+        context "`target` is unset" do
+
+          context "`source` end with [ip]" do
+            let(:event) { LogStash::Event.new("host" => {"ip" => ip}) }
+            let(:options) { common_options.merge({"source" => "[host][ip]"}) }
+
+            it "should use [host] as target" do
```

Comment:
nit: the specs in documentation format can be made more readable if we don't include implementation-specific stuff in the description.

```suggestion
            it "should use source's parent as target" do
```